### ISSUE TITLE
[Bug/Interaction] getItem causes addition of NBT tag when Astral Sorcery is installed

### ISSUE THREAD

**Issue by Aedda** [SquidDev-CC/ccjam-2016#223]
When Astral Sorcery is installed the following code causes an NBT tag to be inserted into the item or items causing them to fail to stack with non-tagged items.
```lua
local p = peripheral.wrap(direction_of_inventory)

p.getItem(n) -- this causes the NBT tag to be inserted into whatever is in slot n of the wrapped inventory
```

The tag inserted is `astralsorcery:{}`

 - Minecraft version: 1.12.2
 - Forge version: 2838
 - CC: Tweaked version: 1.12.2-1.83.1
 - Plethora version: 1.12.2-1.2.1
 - Astral Sorcery version: 1.12.2-1.10.19

**Comment by SquidDev** [comment]
@Nerdpie Any idea what might be causing this? I know you're more familiar with the AS integration than I am. I'm assuming this is due to a converter, but not entirely sure.

**Comment by Nerdpie** [comment]
Alright, I found the culprit.  The converter `ITEM_STACK_TO_CRYSTAL_PROPERTIES` includes this line:
https://github.com/SquidDev-CC/plethora/blob/245d306cb2d3fff1a2fe2af7e8073ec9286107dd/src/main/java/org/squiddev/plethora/integration/astralsorcery/IntegrationAstralSorcery.java#L61

In the Astral Sorcery code that is called there, if Astral doesn't find its own NBT tag, it adds it.
https://github.com/HellFirePvP/AstralSorcery/blob/4f55af9e4bb96d328d5ddbb4bddbe64df5c850e0/src/main/java/hellfirepvp/astralsorcery/common/util/nbt/NBTHelper.java#L53-L63

Potential fixes include:
* replicating AS' NBT parsing ourselves
* extending the list of `instanceof` checks
* calling AS' `hasPersistentData` method first

The last option is probably the cleanest, most straight-forward fix.

**Comment by SquidDev** [comment]
Right, should be fixed in 1.2.2. Sorry for the delay, and thank you @Nerdpie for the quick diagnosis!
